### PR TITLE
Refactoring zoom sync.

### PR DESCRIPTION
**Pull request by peterdeweese**
# Description

Tweaked some rubocop standards after discussion with the group. DRY-ed some code in the zoom sync by making the link columns and operations related to them interated instead of duplicated.

# Motivation and Context

To learn about our sync mechanism and data structures.
To pair with Ben regarding the same and teach about refactoring.
To reduce code smells and get a cleaner lint so we can understand and maintain or improve this code more readily.

Task: 

https://app.asana.com/0/0/1203467706493157/f
https://app.asana.com/0/1201131148207877/1202848766806841/f

### CODE REVIEW

File: app/models/participant_sync_info.rb [hunk 1]
```diff
@@ -47,10 +47,10 @@ def self.primary_key
   module SyncScope
     def self.included(base)
       base.class_eval do
-        # Note: when using a WHERE clause to limit what this scope selects, you need to specify the
+        # NOTE: when using a WHERE clause to limit what this scope selects, you need to specify the
         # the columns on the HerokuConnect::Participant table. See heroku_connect/participant_spec.rb
         # for a list of those
-        scope :live_sf_data, -> {
+        scope :live_sf_data, lambda {
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:53]: Many of these nit-picky changes are from Rubocop.

File: app/models/participant_sync_info.rb [hunk 2]
```diff
@@ -219,13 +219,11 @@ def accelerator_course_role
...
-      RoleConstants::STUDENT_ENROLLMENT
-    elsif role_category == SalesforceConstants::RoleCategory::TEACHING_ASSISTANT
-      RoleConstants::TA_ENROLLMENT
-    else
-      nil # Fellows (and others) aren't in the LC Playbook
+    case role_category
+    when SalesforceConstants::RoleCategory::LEADERSHIP_COACH then RoleConstants::STUDENT_ENROLLMENT
+    when SalesforceConstants::RoleCategory::TEACHING_ASSISTANT then RoleConstants::TA_ENROLLMENT
     end
+    # Fellows (and others) aren't in the LC Playbook, so nil is returned.
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:226]: nil is automatically returned from an unsatisfied if-else.

File: app/models/participant_sync_info.rb [hunk 3]
```diff
@@ -266,6 +264,16 @@ def add_to_honeycomb_span(suffix = nil)
     attributes.each_pair { |attr, value| Honeycomb.add_field("participant_sync_info.#{attr}#{suffix}", value.to_s) }
   end
 
+  # 1 or 2 to match #zoom_meeting_id_#{index}
+  def zoom_meeting_id(index)
```
Comment (peterdeweese) [inline app/models/participant_sync_info.rb:268]: This is the linchpin that allows for iteration. Note the argument error.

File: spec/models/participant_sync_info_spec.rb [hunk 4]
```diff
@@ -26,37 +26,63 @@
...
+
+      it { is_expected.to eq '12' }
+    end
+
+    context 'with 2' do
+      let(:index) { 2 }
+
+      it { is_expected.to eq '13' }
+    end
+  end
```
Comment (tomlovett) [inline spec/models/participant_sync_info_spec.rb:66]: nice

**Review by tomlovett — APPROVED** [review]
Great work! LGTM

File: app/models/participant_sync_info.rb [hunk 5]
```diff
@@ -412,9 +416,7 @@ def requires_cohort_schedule?
     # skip this Participant if there are no changes.
     def save_successful_sync!
       unless should_sync?
-        raise RuntimeError.new(
-          "Cannot save a ParticipantSyncInfo for #{@new_sync_info.sfid} b/c should_sync? is false."
-        )
+        raise "Cannot save a ParticipantSyncInfo for #{@new_sync_info.sfid} b/c should_sync? is false."
```
[the remark this answers is not in the document]
Comment (peterdeweese) [reply app/models/participant_sync_info.rb:419]: The default `raise` class if nothing is specified is a `RuntimeError`!

**Pull request merged**